agent/agentserver/server.go

```diff
@@ -146,6 +155,24 @@ func (s *Server) deleteBlobHandler(w http.ResponseWriter, r *http.Request) error
 	return nil
 }
 
+// preloadTagHandler triggers docker daemon to download specified docker image.
+func (s *Server) preloadTagHandler(w http.ResponseWriter, r *http.Request) error {
+	tag, err := httputil.ParseParam(r, "tag")
+	if err != nil {
+		return err
+	}
+	parts := strings.Split(tag, ":")
+	if len(parts) != 2 {
+		return handler.Errorf("failed to parse docker image tag")
+	}
+	if err := s.dockerCli.ImagePull(
+		context.Background(), r.Host, parts[0], parts[1]); err != nil {
```

Comment:
Won't `r.Host` always be localhost?

Reply:
Also need port, which is only available through r.Host